### PR TITLE
add show and hide for tags on homepage

### LINKED ISSUE #19: Make tag selector more useful

### PR DESCRIPTION
**Pull request by claireynz**
Close #19.

### CODE REVIEW

**Comment by loicteixeira** [comment]
This looks great @claireynz 🎉
tags

@claireynz, @thibaudcolas: Tags are ordered by name which makes sense when they are all shown. Do you think we should order them by number of posts now, so the most popular are shown without the need to click "show more"? If so, do you think the tag should include the number of posts in it? FWIW, I'm happy to do the back-end part of this.

File: core/templates/core/includes/tags.html [hunk 1]
```diff
@@ -13,4 +14,4 @@
         {{ tag }}
     </a>
 {% endfor %}
-</div>
\ No newline at end of file
+</div>
```
Comment (loicteixeira) [inline core/templates/core/includes/tags.html]: I don't think this file is used. If so, it should either be deleted, or the tag logic moved from `core/templates/core/home_page.html` to this file.

**Comment by thibaudcolas** [comment]
Updated with a few changes:

- Moved the JS to a separate file.
- Cleaned up the HTML and CSS a bit (not related to those specific changes).
- The JS performance does not depend on the number of tags anymore.
- If JS is disabled, display all tags.
- Add ARIA attributes for screen readers to know whether the tags are displayed or not.

@loicteixeira @claireynz could you review?

For your point about ordering Loïc – I think it makes perfect sense to order the tags by number of sites for each. Not sure whether we would want to display that count on the tag though. Would you want to do this in this PR or in another one?

**Comment by loicteixeira** [comment]
I'm happy to do this part of this PR if you don't mind waiting for a day or two.

**Comment by thibaudcolas** [comment]
Sounds good 🙂

**Comment by loicteixeira** [comment]
Done. I also tackled #44 as I was there.

It will need some styling though. Leaving it to you @claireynz.
tags with count

**Comment by thibaudcolas** [comment]
Cool! @bwjwright you've worked on this project before I think, would you have some thoughts on what this should look like? Should there be a count at all?

File: core/models.py [hunk 2]
```diff
@@ -87,7 +88,7 @@ def get_context(self, request):
         # Filter by tag
         tag = request.GET.get('tag')
         if tag:
-            pages = pages.filter(tags__name__iexact=tag)
+            pages = pages.filter(tags__slug__iexact=tag)
```
Comment (thibaudcolas) [inline core/models.py]: I'm not sure I understand this exact change, is this related to #44?

Comment (loicteixeira) [inline core/models.py]: Yes. The link is [now generated](https://github.com/springload/madewithwagtail/pull/64/commits/f62c2eb1641363711321fbde975951a9c77c8572#diff-0e958c02386cf27e2da09d8ad41da5f1R33) from the `slug`, not the `name` so the back-end has to be updated accordingly. It avoids having funky URL like `/?tag=Social%20Media`.

FWIW, I forgot to change it on the individual site page and did it separately in 50d1f1c0ac703163f7a0086f16d10a4b3c8f739a.

**Comment by bwjwright** [comment]
@thibaudcolas Hello! The numbers are an interesting idea and it's cool to see how many projects are behind a tag instead of clicking into a tag and only finding 1 thing (for example).

You could try put a circle or some other visual behind the number but it might be too much, by default I think it does the job as is. It probably needs some exploration if we want to do something different though.
